### PR TITLE
[PLAT-11] feat testing : Added AlertMessage component

### PR DESCRIPTION
**Pull request by NamrataAjmeri**
## Scope
- [x] Enhancement: backwards-compatible functionality added
- [ ] Bug-Fix / Patch: backwards-compatible bug fix / revision
- [ ] RFC: request for comments; discussion only

## Context
Added alert design

## Summary of Changes
Added AlertMessage component to update alert design.

## Additional Considerations
Any additional consequences, side effects, uncertainties stemming from the changes in this PR.

## Pre-Merge Checklist
- [ ] Changelog entry? (Summary in `/changelog/<section>/<issue_number>.md`)
- [ ] Linked to [Jira issue](https://support.atlassian.com/bitbucket-cloud/docs/use-smart-commits/) ?
- [ ] Labels tagged?

### CODE REVIEW

File: src/AlertMessage/AlertMessage.stories.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+Default.args = {
+  type: "warning",
+  simplified: false,
+};
+
+export default {
+  title: "AlertMessage",
+  component: AlertMessage,
+  parameters: {
+    componentSubtitle: "",
```
Comment (johnisr) [inline src/AlertMessage/AlertMessage.stories.js:22]: Screen Shot 2022-02-22 at 11 01 40 AM

This text is just what shows up in the Docs tab for the component (e.g. `A wrapper component...` in the screenshot above), if there's more info like what it's used for, a short description, etc. you can add it here or just leave this out

File: src/AlertMessage/AlertMessage.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import classNames from "classnames/bind";
+import styles from "./AlertMessage.module.scss";
+import AlertMessageType from "../types/AlertMessage";
+import FontAwesome from "react-fontawesome";
+
+const cx = classNames.bind(styles);
+
+const AlertMessage = ({
+  type,
+  simplified = false,
```
Comment (johnisr) [inline src/AlertMessage/AlertMessage.tsx:11]: is simplified the toggle between low priority alerts and high priority alerts? might be better to to have this as `hasBackground` instead, or something more recognizable what the effect will be.

File: src/AlertMessage/AlertMessage.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        simplified
+          ? `${cx("alertMessageContainer")}`
+          : `${cx("alertMessageContainer", `alertMessageContainer--${type}`)}`
+      }
+      role="alert-message"
+    >
+      <div className={cx("informationContainer")}>
+        <div className={cx("icon")}>{getIconByType(type)} </div>
+        <div className={cx("text")}>{children}</div>
+      </div>
```
Comment (johnisr) [inline src/AlertMessage/AlertMessage.tsx:53]: The design also has a close icon like in `Alert` as its part of the design instructions. e.g.
```
      {onClose && (
        <button
          ...
          onClick={onClose}
        >
          <FontAwesome name="times fa fa-fw" />
        </button>
      )}
```

File: src/Form/FormInputs/FormInputWrapper.tsx [hunk 4]
```diff
@@ -66,7 +67,11 @@ export const FormInputWrapper = ({
               isError={!!errorMessage}
             />
 
-            <FormError message={!isDisabled ? errorMessage : null} />
+            {!isDisabled && !!errorMessage ? (
+              <AlertMessage type="alert" simplified={true}>
+                {errorMessage}
+              </AlertMessage>
+            ) : null}
```
Comment (johnisr) [inline src/Form/FormInputs/FormInputWrapper.tsx:74]: If this is intended to replace FormError, make sure to delete the component and its styles.

File: src/AlertMessage/AlertMessage.module.scss [hunk 5]
```diff
@@ -0,0 +1,49 @@
+.alertMessageContainer {
+  border-radius: 6px;
+  padding: 15px 0px;
```
Comment (johnisr) [inline src/AlertMessage/AlertMessage.module.scss:3]: Make sure no padding is added when `simplified`. It's acceptable when doing the `non-simplified` version as it styles the coloured background. components are a lot easier to use when there's no inherent padding for the outermost container and the spacing can just be adjusted the developers who use it.

For example, having this when it's simplified makes the spacing between the FormInputs way bigger than intended. it overwrites the intended 8px spacing between the input and the error message and almost doubles it. 

Screen Shot 2022-02-22 at 11 39 41 AM

File: src/types/AlertMessage/index.tsx [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import { ReactNode } from "react";
+
+export type MessageType = "alert" | "warning" | "info";
+
+type AlertMessageType = {
+  /** Describes if the message displayed is info, alert or warning */
+  type: MessageType;
+  /**  */
+  simplified: boolean;
```
Comment (johnisr) [inline src/types/AlertMessage/index.tsx:9]: Needs a description of what it is.

File: src/AlertMessage/AlertMessage.tsx (hunk 2 above)
Comment (NamrataAjmeri) [reply src/AlertMessage/AlertMessage.tsx:11]: Correct, it's a toggle. The name is suggested by Chung. @Chung Should I implement John's suggestion ?

Comment (chungcareer) [reply src/AlertMessage/AlertMessage.tsx:11]: Its less of a toggle but a type you would add it. By default it should already have a background.

This is how I think of it 
```
<AlerMessage simplified />
```

File: src/AlertMessage/AlertMessage.tsx [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+      case "warning":
+        return (
+          <FontAwesome
+            name="exclamation-circle"
+            className={`${cx("warningicon")}`}
+          />
+        );
+    }
+    return null;
+  };
```
Comment (chungcareer) [inline src/AlertMessage/AlertMessage.tsx:40]: This should be a component ie) <AlertMessageIcon type="alert" />

File: src/AlertMessage/AlertMessage.tsx [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+  return (
+    <div
+      className={
+        simplified
+          ? `${cx(
+              "alertMessageContainer",
+              "alertMessageContainer--simplified"
+            )}`
+          : `${cx("alertMessageContainer", `alertMessageContainer--${type}`)}`
+      }
```
Comment (johnisr) [inline src/AlertMessage/AlertMessage.tsx:25]:
```suggestion
      className={cx(
        "alertMessageContainer",
        simplified
          ? "alertMessageContainer--simplified"
          : `alertMessageContainer--${type}`
      )}
```

**Review by johnisr — APPROVED** [review]

**Comment by johnisr** [comment]
Remember to create a new PR for the build file and release so you can use the changes here in your 360 PR.

**Pull request merged**